uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -247,9 +264,56 @@ define([
          * @param {string} key - The property name
          */
         toAPIFilter: function (result, __, key) {
-          if (!_.includes(['balance_change', 'dates', 'comments'], key)) {
+          if (!_.includes(['balance_change', 'dates', 'comments', 'uploader', 'files'], key)) {
             result[key] = this[key];
           }
+        },
+
+        /**
+        * Gets file attachments associated with this leave request
+        *
+        * @return {Promise}
+        */
+        getAttachments: function () {
+          return LeaveRequestAPI.getAttachments(this.id)
+            .then(function (attachments) {
+              this.files = attachments;
+            }.bind(this));
+        },
+
+        /**
+        * Deletes the given attachment from local files array. It sets a flag
+        * to delete in final update call. This can be used by the client to set
+        * file for deletion.
+        */
+        deleteAttachment: function (attachmentID) {
+          this.files = this.files || [];
```

Comment:
Why is this necessary? `files` should be initialized always when calling `init()` on the instance